libethereum/ValidationSchemes.cpp

```diff
@@ -118,37 +118,42 @@ void validateAccountObj(js::mObject const& _obj)
                 {c_wei, {{js::str_type}, JsonFieldPresence::Optional}},
                 {c_balance, {{js::str_type}, JsonFieldPresence::Optional}}});
     }
-    else if (_obj.size() == 1)
-    {
-        // A genesis account with only balance set
-        if (_obj.count(c_balance))
-            requireJsonFields(_obj, "validateAccountObj",
-                {{c_balance, {{js::str_type}, JsonFieldPresence::Required}}});
-        else
-            requireJsonFields(_obj, "validateAccountObj",
-                {{c_wei, {{js::str_type}, JsonFieldPresence::Required}}});
-    }
     else
     {
-        if (_obj.count(c_codeFromFile))
+        // Extendable account description. Check typo errors and fields type.
+        requireJsonFields(_obj, "validateAccountObj",
+            {{c_code, {{js::str_type}, JsonFieldPresence::Optional}},
+                {c_nonce, {{js::str_type}, JsonFieldPresence::Optional}},
+                {c_storage, {{js::obj_type}, JsonFieldPresence::Optional}},
+                {c_balance, {{js::str_type}, JsonFieldPresence::Optional}},
+                {c_wei, {{js::str_type}, JsonFieldPresence::Optional}},
+                {c_codeFromFile, {{js::str_type}, JsonFieldPresence::Optional}}});
+
+        // At least one field must be set
+        if (!_obj.count(c_code) && !_obj.count(c_nonce) && !_obj.count(c_storage) &&
```

Comment:
```suggestion
        if (_obj.empty())
```

Or whatever the equivalent of this is in the used JSON library.

Reply:
Yeah, wrong fields should be checked in verify json above.